pkg/api/v1alpha1/cluster.go

```diff
@@ -832,36 +834,45 @@ func validateCPUpgradeRolloutStrategy(clusterConfig *Cluster) error {
 		return nil
 	}
 
-	if cpUpgradeRolloutStrategy.Type != "RollingUpdate" && cpUpgradeRolloutStrategy.Type != "InPlace" {
+	switch cpUpgradeRolloutStrategy.Type {
+	case rollingUpdate:
+		if cpUpgradeRolloutStrategy.RollingUpdate.MaxSurge < 0 {
+			return fmt.Errorf("ControlPlaneConfiguration: maxSurge for control plane cannot be a negative value")
+		}
+		if cpUpgradeRolloutStrategy.RollingUpdate.MaxSurge > 1 {
+			return fmt.Errorf("ControlPlaneConfiguration: maxSurge for control plane must be 0 or 1")
+		}
+	case inPlaceUpgrade:
+		if clusterConfig.Spec.DatacenterRef.Kind != TinkerbellDatacenterKind {
+			return fmt.Errorf("ControlPlaneConfiguration: 'InPlace' upgrade rollout strategy type is only supported on the Tinkerbell provider")
```

Comment:
Do we say Tinkerbell provider officially or BareMetal provider?

Reply:
It looks like we do say "Bare Metal". I'll update!